src/mo_values/prim.ml

```diff
@@ -52,17 +52,21 @@ Wrapping numeric conversions are all specified uniformly by going through bigint
 (* Trapping conversions (the num_conv_t1_t2 prim used in prelude/prelude.ml) *)
 let num_conv_trap_prim t1 t2 =
   let module T = Type in
-  match (t1, t2) with
-  | T.Nat, (T.Nat8|T.Nat16|T.Nat32|T.Nat64)
-  | T.Int, (T.Int8|T.Int16|T.Int32|T.Int64)
-  | (T.Nat8|T.Nat16|T.Nat32|T.Nat64), T.Nat
-  | (T.Int8|T.Int16|T.Int32|T.Int64), T.Int
+  match t1, t2 with
+  | T.Nat, T.(T.Nat8|Nat16|Nat32|Nat64)
```

Comment:
`T.Nat8` is  necessary to suppress a warning.

Reply:
If `T.Nat8` is not supposed to be handled but only here to suppress a warning I think it would be better to turn that case into a panic.

Reply:
It is supposed to be there, I don't want to change the semantics. It is purely about type ambiguity. There are two constructors in different types called `Nat8`, and the compiler can disambiguate them, but warns. This kind-of hits it on the nose. No idea why the `T.(...)` scope extension won't give `T.Nat8` a preferred status.